app/models/manageiq/providers/redhat/infra_manager/event_parsing/strategies/v4.rb

```diff
@@ -27,9 +27,20 @@ def self.event_to_hash(event, ems_id = nil)
         :vm_ems_ref          => vm_ref,
         :host_ems_ref        => ems_ref_from_object_in_event(event.host),
         :ems_cluster_ems_ref => ems_ref_from_object_in_event(event.cluster),
+        :vm_location         => vm_location(event.data_center, vm_ref)
       }
     end
 
+    def self.vm_location(dc, vm_ref)
+      uid_ems = ManageIQ::Providers::Redhat::InfraManager.extract_ems_ref_id(vm_ref)
+      location = "#{uid_ems}.ovf"
+      return location if dc.blank?
+
+      dc_ref = ems_ref_from_object_in_event(dc)
+      dc_uid = ManageIQ::Providers::Redhat::InfraManager.extract_ems_ref_id(dc_ref)
+      File.join('/rhev/data-center', dc_uid, 'mastersd/master/vms', uid_ems, location)
```

Comment:
I tried to look for that path on my env and could find any data under /rhev/data-center/2c63ebdf-2784-4129-aa77-d80fa6571c18/mastersd/master/vms/ (the folder exists, but it is empty)

The vms are located under a specific storage domain.
@danielerez Could you have a look and advise ?

Reply:
@masayag The patch that I started to build here is based on [this](https://github.com/ManageIQ/manageiq/blob/master/app/models/vm_or_template.rb#L1227). This is exactly how the code is checking it.